### ISSUE TITLE
The wildcard can't be used by the C/C++ compiler recommanded!

### ISSUE THREAD

**Issue by WangShiSuiFeng24** [microsoft/vscode-cpptools#13096]
When I browsed the doc of GCC on Windows，I found that I could modify my tasks.json file to build multiple C++ files by using an argument like "${workspaceFolder}/*.cpp" instead of "${file}". But there is a fatal error when I run my program. The information is that *.cpp is invalid argument. 

What is interesting is that my program runs sucessfully if I use another compiler downloaded from https://winlibs.com/.
![Image](https://github.com/user-attachments/assets/aa9facee-2dea-4344-8351-cedd0c8b3935)

**Comment by browntarik** [comment]
Could you provide more information about this compiler you are using? What is the exact path of "g++" in the command line?

 Does this repro when you use the full path to the ucrt version of the compiler (i.e "C:/msys64/ucrt64/bin/g++.exe") in the command line?

**Comment by WangShiSuiFeng24** [comment]
Yes, g++ means D:\msys64\ucrt64\bin\g++.exe which is downloaded by msys2 that vscode document recommended.
I have tried many times in the command line and also in tasks.json.
You can download the msys2 and then input the command "pacman -S --needed base-devel mingw-w64-ucrt-x86_64-toolchain
" according to https://code.visualstudio.com/docs/cpp/config-mingw.
Finally, at windows10 environment, you can try the command "g++ *.cpp -o filename.exe". There will be a fatal error.

**Comment by browntarik** [comment]
If the issue reproduces when you just use the command line, then the issue may have to do with your shell and not our extension. I am not able to reproduce your behavior using the same compiler. 

It wasn't clear based on your response if you have tried compiling with the full path. Can you confirm that doesn't work?

Did you ensure the ucrt compiler was added to your PATH environment variable?

**Comment by WangShiSuiFeng24** [comment]
No，the issue also happened in tasks.json. The shell is powershell of windows10. 
I did add the ucrt compiler to the PATH environment variable. I can be sure that I also tried full path command compile. That made a fatal error.
I have also communicated with some friends on the internet. They also found there is a fatal error.
pwsh
![Image](https://github.com/user-attachments/assets/4a32fa87-21db-47f5-ac4c-efa1675b6556)
cmd
![Image](https://github.com/user-attachments/assets/4eaa2aa9-b1ca-4dfb-b9cf-630251061665)
tasks.json
![Image](https://github.com/user-attachments/assets/4cf0b1d3-c1db-401b-99e5-a2d4371695e4)

**Comment by SainoNamkho** [comment]
> If the issue reproduces when you just use the command line, then the issue may have to do with your shell and not our extension. I am not able to reproduce your behavior using the same compiler.
> 
> It wasn't clear based on your response if you have tried compiling with the full path. Can you confirm that doesn't work?
> 
> Did you ensure the ucrt compiler was added to your PATH environment variable?

This issue is more about documentaton rather than cpptools itself. Maybe https://github.com/microsoft/vscode-docs/issues is the better place to report it.

https://code.visualstudio.com/docs/cpp/config-mingw recommends using gcc from msys2, which was not configured with `--enable-mingw-wildcard`, thus the wildcard expansion only works in msys shell. not the integrated terminal where `tasks.json` is used.

However gcc from winlibs is configured with `--enable-mingw-wildcard`.

**Comment by truongan012** [comment]
It's disabled by msys2 as default ;( 
https://www.msys2.org/news/#2024-11-03-disabling-mingw-w64-wildcard-support-by-default

I dont know how to enable it again!?

**Comment by browntarik** [comment]
> > If the issue reproduces when you just use the command line, then the issue may have to do with your shell and not our extension. I am not able to reproduce your behavior using the same compiler.
> > It wasn't clear based on your response if you have tried compiling with the full path. Can you confirm that doesn't work?
> > Did you ensure the ucrt compiler was added to your PATH environment variable?
> 
> This issue is about documentation rather than cpptools itself. Maybe https://github.com/microsoft/vscode-docs/issues is the better place to report it.
> 
> https://code.visualstudio.com/docs/cpp/config-mingw recommends using gcc from msys2, which was not configured with `--enable-mingw-wildcard`, thus the wildcard expansion only works in msys bash shell, not the integrated terminal where `tasks.json` is used.
> 
> However gcc from winlibs is configured with `--enable-mingw-wildcard`.

@SainoNamkho Thank you for providing this additional context, it looks like MSYS2 decided to change their compiler behavior at the end of the year. We will update our documentation to reflect this change in behavior.

**Comment by WangShiSuiFeng24** [comment]
This issue seems to be over. Thank you all for answering my questions.
